src/com/egeniq/widget/SpannableTextView.java

```diff
@@ -64,8 +108,55 @@ public void append(int style, String text) {
     }
 
     /**
-     * Updates the TextView with markup text from the _markupText collection.
-     * Each text segment is appended to the last.
+     * Overwrites the current content of this SpannableTextView with new content. Update() must be called to update the TextView visually.
+     *
+     * @param span the DynamicDrawableSpan to be applied
+     */
+    public void set(DynamicDrawableSpan span) {
+        clear();
+        append(span);
+    }
+
+    /**
+     * Appends a DynamicDrawableSpan to the SpannableTextView. Update() must be called to update the TextView visually.
+     *
+     * @param span
+     */
+    public void append(DynamicDrawableSpan span) {
+        _markupList.add(new MarkupData(" ", span)); //The space will be styled with an icon
+        _markupText += " ";
```

Comment:
Image wont be displayed if I don't add a character as placeholder for the image. I agree it's a bit dirty. Open for suggestions.